app/controllers/api/v1/profiles_controller.rb

```diff
@@ -0,0 +1,61 @@
+class Api::V1::ProfilesController < Api::V1::ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    if user_signed_in?
```

Comment:
I think this (`user_signed_in?`) is a devise method and you don't have Devise here. Also, it's better practice to use `create` instead of `new`